build/test-threadSafety.ps1

```diff
@@ -0,0 +1,35 @@
+param(
+  [Parameter(Mandatory=$true)][string]$coyoteVersion="",
+  [Parameter(Mandatory=$true)][string]$targetFramework,
+  [Parameter(Mandatory=$true)][string]$methodNames
+)
+
+$rootDirectory = Split-Path $PSScriptRoot -Parent
+
+Write-Host "Install Coyote CLI."
+dotnet tool install --global Microsoft.Coyote.CLI
+
+Write-Host "Build OpenTelemetry.Tests project."
+dotnet build $rootDirectory\test\OpenTelemetry.Tests\OpenTelemetry.Tests.csproj
+
+$artifactsPath = Join-Path $rootDirectory "test\OpenTelemetry.Tests\bin\Debug\$targetFramework"
+
+Write-Host "Generate Coyote rewriting options JSON file."
+$assemblies = Get-ChildItem $artifactsPath -Filter OpenTelemetry*.dll | ForEach-Object {$_.Name}
+
+$RewriteOptionsJson = @{}
+[void]$RewriteOptionsJson.Add("AssembliesPath", $artifactsPath)
+[void]$RewriteOptionsJson.Add("Assemblies", $assemblies)
+$RewriteOptionsJson | ConvertTo-Json -Compress | Set-Content "$rootDirectory\test\OpenTelemetry.Tests\rewrite.coyote.json"
+
+Write-Host "Run Coyote rewrite."
+coyote rewrite $rootDirectory\test\OpenTelemetry.Tests\rewrite.coyote.json
+
+Write-Host "Execute re-written binary."
+$Output = dotnet test $artifactsPath\OpenTelemetry.Tests.dll --filter $methodNames
+
+Write-Host "Coyote test output:"
```

Comment:
What output is expected here? `dotnet test` should return either `0` or `1`.

Reply:
The re-written binary can be executed as normal XUnit test -
so the dotnet test command will fail if the things we are asserting on it not true.

Reply:
What do you expect the value for `$Output` to be?

Reply:
just wanted to see the coyote logs when I was doing prototyping -
I can remove the console print given the assumption that if the test fail devs will no matter debug locally

Reply:
`$Output` variable will either have a value of 0 or 1.

There is no need to split and parse 0 or 1 😀

```csharp
foreach ($line in $($Output -split "`r`n"))
{
    Write-Host $line
}
```

Reply:
ah gotcha! 🗡️